# ANALISIS DE SALARIOS VIGENTES A PARTIR DEL 1o. DE FEBRERO PARA EL ESTADO DE GUANAJUATO 2019

In [6]:
# Impotamos las lbrerias necesarias
import numpy as np
import math as mt
import matplotlib.pyplot as plt
import pandas as pd
import requests
import os

### SALARIOS GUANAJUATO

In [7]:
#SE CARGAN LOS DATOS DE LOS SALARIOS

path = f'/home/luiscantero/anadat/Archivos/Proyecto1/Salarios/tabulador_pe2019.xls'
tabulador = pd.read_excel(path, header=9)


In [9]:
tabulador.head()

NIVEL TIPO                            PUESTO   SUELDO BASE  \
0     22  NaN           GOBERNADOR/A DEL ESTADO  59804.327930   
1     20  NaN          SECRETARIO/A DE ESTADO A  49025.945517   
2     19  NaN  DIRECTOR/A GENERAL PARAESTATAL A  40685.951590   
3     19  NaN          SECRETARIO/A DE ESTADO B  40685.951590   
4     18  NaN          SECRETARIO/A DE ESTADO C  37083.708634   

   CUOTAS SEGURIDAD SOCIAL  PREVISIÓN SOCIAL  AYUDA X SERVICIOS  \
0                  5843.07            1913.1              25897   
1                  5843.07            1913.1              22783   
2                  5843.07            1913.1              18805   
3                  5843.07            1913.1              18805   
4                  5843.07            1913.1              16756   

   APOYO FAMILIAR  GRATIFICACIÓN QUINCENAL  SUELDO MENSUAL BRUTO  \
0    64294.837780                 75373.99         233126.325711   
1    48858.839699                 67187.62         195611.575216   
2    41397.705456                 49230.82         157875.647046   
3    41397.705456                 49230.82         157875.647046   
4    35924.238404                 42651.90         140172.017038   

   IMPTO MENSUAL    ISSEG  ISSSTE  SUELDO MENSUAL NETO PRIMA VACACIONAL  \
0   69584.321342  5082.66  760.41        157698.934369  10 Días por Año   
1   56829.306173  5082.66  760.41        132939.199042  10 Días por Año   
2   43999.090596  5082.66  760.41        108033.486450  10 Días por Año   
3   43999.090596  5082.66  760.41        108033.486450  10 Días por Año   
4   37979.856393  5082.66  760.41         96349.090645  10 Días por Año   

         AGUINALDO  
0  45 Días por Año  
1  45 Días por Año  
2  45 Días por Año  
3  45 Días por Año  
4  45 Días por Año

### Codigos para las preguntas de salarios

¿Cuál es promedio de las contribuciones del sueldo de un trabajador en esta dependencia?

In [10]:
tabulador.describe()['SUELDO MENSUAL BRUTO']['mean']

49467.37593510606

In [11]:
tabulador.describe()['SUELDO MENSUAL NETO']['mean']

35829.35357437967

¿Media y Moda de las contribuciones del sueldo de un trabajador ? Rango interquartil de la distribución.

In [12]:
tabulador['SUELDO MENSUAL NETO'].mode()

0    8294.835952
1    9609.786902
dtype: float64

In [13]:
tabulador.describe()['SUELDO MENSUAL NETO'].loc['25%':'75%']

25%    10289.592141
50%    22078.225282
75%    50363.213596
Name: SUELDO MENSUAL NETO, dtype: float64

¿Cuál es la dispersión de sueldos?

In [14]:
median_sneto = tabulador['SUELDO MENSUAL NETO'].median()
print(median_sneto)

22078.22528193275


¿Cuál es el sueldo base máximo?

In [15]:
tabulador.describe()['SUELDO BASE'].loc['max']

59804.3279302758

¿Cuál es el sueldo base mínimo?

In [16]:
tabulador.describe()['SUELDO BASE'].loc['min']

3080.4

Porcentaje relativo de cada una de las componentes del sueldo mensual neto en cada uno de los puestos

In [17]:
tabulador['SUELDO NETO (%)'] = 100 * tabulador['IMPTO MENSUAL']/np.sum(tabulador['IMPTO MENSUAL'])

In [18]:
tabulador.loc[:,['PUESTO','SUELDO NETO (%)']].head()

PUESTO  SUELDO NETO (%)
0           GOBERNADOR/A DEL ESTADO         7.944592
1          SECRETARIO/A DE ESTADO A         6.488325
2  DIRECTOR/A GENERAL PARAESTATAL A         5.023471
3          SECRETARIO/A DE ESTADO B         5.023471
4          SECRETARIO/A DE ESTADO C         4.336242

In [19]:
ruta = "porc_rel.csv"
tabulador.loc[:,['PUESTO','SUELDO NETO (%)']].to_csv(ruta)

¿De cuánto dinero es la diferencia entre el sueldo neto mayor (gobernador del estado) y el sueldo de otros empleados (por ejemplo, alguien de nivel 1, que es el nivel más bajo en la tabla)?


In [20]:
tabulador['DIFERENCIA'] = np.max(tabulador['SUELDO MENSUAL NETO']) - tabulador['SUELDO MENSUAL NETO']

In [21]:
tabulador.loc[:,['PUESTO','SUELDO MENSUAL NETO','DIFERENCIA']].head()

PUESTO  SUELDO MENSUAL NETO    DIFERENCIA
0           GOBERNADOR/A DEL ESTADO        157698.934369      0.000000
1          SECRETARIO/A DE ESTADO A        132939.199042  24759.735327
2  DIRECTOR/A GENERAL PARAESTATAL A        108033.486450  49665.447919
3          SECRETARIO/A DE ESTADO B        108033.486450  49665.447919
4          SECRETARIO/A DE ESTADO C         96349.090645  61349.843724

In [22]:
ruta = "diferencia.csv"
tabulador.loc[:,['PUESTO','SUELDO MENSUAL NETO','DIFERENCIA']].to_csv(ruta)

¿Qué porcentaje se da de apoyo en sus diferentes modalidades para cada uno?¶ 

In [23]:
apoyo = []
for i in range(len(tabulador.PUESTO)):
    x = np.sum(tabulador.iloc[i][['CUOTAS SEGURIDAD SOCIAL','PREVISIÓN SOCIAL','AYUDA X SERVICIOS','APOYO FAMILIAR','GRATIFICACIÓN QUINCENAL']])
    apoyo.append(x)
    

tabulador['APOYO TOTAL'] = apoyo
tabulador['APOYO TOTAL (%)'] = 100*tabulador['APOYO TOTAL']/np.sum(apoyo)

In [24]:
tabulador.loc[:,['PUESTO','APOYO TOTAL','APOYO TOTAL (%)']].head()

PUESTO    APOYO TOTAL  APOYO TOTAL (%)
0           GOBERNADOR/A DEL ESTADO  173321.997780         6.261298
1          SECRETARIO/A DE ESTADO A  146585.629699         5.295441
2  DIRECTOR/A GENERAL PARAESTATAL A  117189.695456         4.233505
3          SECRETARIO/A DE ESTADO B  117189.695456         4.233505
4          SECRETARIO/A DE ESTADO C  103088.308404         3.724090

In [25]:
ruta = "apoyo.csv"
tabulador.loc[:,['PUESTO','APOYO TOTAL','APOYO TOTAL (%)']].to_csv(ruta)

¿Cuanto dinero es gastado en es ISSSTE entre todos los puestos?

In [26]:
np.sum(tabulador['ISSSTE'])

29411.26871446277

In [27]:
np.sum(tabulador['ISSEG'])

172122.23464139525

##### -¿Cuál es promedio de las contribuciones del sueldo de un trabajador en esta dependencia?

Sueldo bruto: $ 49467.37

Sueldo neto: $35829.35

##### -¿Media y Moda de las contribuciones del sueldo de un trabajador ? Rango interquartil de la distribución.

Moda: 8294.83, 9609.78

Rangos intercuartiles

25%    $10289.59

50%    $22078.22

75%    $50363.21

##### -¿Cuál es la dispersión de sueldos?

##### -¿Cuál es el sueldo base máximo?

$ 59804.32

##### -¿Cuál es el sueldo base mínimo?

$ 3080.4

##### -Porcentaje relativo de cada una de las componentes del sueldo mensual neto en cada uno de los puestos

##### -¿De cuánto dinero es la diferencia entre el sueldo neto mayor (gobernador del estado) y el sueldo de otros empleados (por ejemplo, alguien de nivel 1, que es el nivel más bajo en la tabla)?


##### -¿Qué porcentaje se da de apoyo en sus diferentes modalidades para cada uno?

##### -¿Cuanto dinero es gastado en es ISSSTE entre todos los puestos?

ISSSTE =  $29411.26

ISSEG =  $172122.23